manifests/params.pp

```diff
@@ -184,6 +184,16 @@
   }
 
   case $::operatingsystem {
+    'Debian': {
+      case $::operatingsystemmajrelease {
```

Comment:
Hi @vorlock, thanks for this patch!
It's possible to use something a bit more dynamic ? (it's just a suggestion ;-))
```puppet
if versioncmp($::operatingsystemmajrelease, '8') >= 0 {
  $service_provider = pick($service_provider, 'systemd')
}
```
(the default is set to undef)

And if possible, a rspec test would be nice :-) (thanks in advance!)